File: WingedPlate.Domain/Entities/RecipeEntity.cs
namespace WingedPlate.Domain.Entities;

public class RecipeEntity : EntityBase
{
    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public byte[] Image { get; set; } = [];

    public double PreparationTime { get; set; }

    public int Calories { get; set; }

    public int CaloriesByPortion { get; set; }

    public decimal ProteinByPortion { get; set; } 

    public decimal FatByPortion { get; set; }

    public decimal CarbohydratesByPortion { get; set; }

    public int PortionsCount { get; set; }  

    public int LevelId { get; set; } = 0;

    public LevelEntity Level { get; set; } = new(); 

    public List<string> Ingredients { get; set; } = [];

    public List<string> Instructions { get; set; } = [];

    public List<TagEntity> Tags { get; set; } = [];

    public List<MealTypeEntity> MealTypes { get; set; } = [];

    public List<CategoryEntity> Categories { get; set; } = [];

    public List<CommentEntity> Comments { get; set; } = [];
}

File: WingedPlate/src/entities/recipe.entity.ts
import { CategoryEntity } from "./category.entity"
import { CommentEntity } from "./comment.entity"
import { LevelEntity } from "./level.entity"
import { MealTypeEntity } from "./mealType.entity"
import { TagEntity } from "./tag.entity"

export class RecipeEntity
{
    public title: string = "";

    public description: string = "";

    public image: Uint8Array = new Uint8Array();

    public preparationType: number = 0.0;

    public calories: number = 0;

    public caloriesByPortion: number = 0;

    public ProteinByPortion: number = 0.0;

    public FatByPortion: number = 0.0;

    public CarbohydratesByPortion: number = 0.0;

    public PortionsCount: number = 0.0;

    public levelId: number = 0;

    public level: LevelEntity = new LevelEntity();

    public ingredients: Array<string> = [];

    public instructions: Array<string> = [];

    public tags: Array<TagEntity> = [];

    public mealTypes: Array<MealTypeEntity> = [];

    public categories: Array<CategoryEntity> = [];

    public comments: Array<CommentEntity> = [];
}